lib/postprocessing/image.js

```diff
@@ -94,113 +93,50 @@ function processBasicImageOperations(img, instructions) {
         img.units('pixelsperinch');
         if (typeof instructions.convertToDpi === 'object') {
             const { xdpi, ydpi } = instructions.convertToDpi;
-            img = img.resample(xdpi, ydpi);
+            img.resample(xdpi, ydpi);
         } else {
-            img = img.resample(instructions.convertToDpi, instructions.convertToDpi);
+            img.resample(instructions.convertToDpi, instructions.convertToDpi);
         }
     }
-
-    console.log('Processing rendition... Done!');
-    return img;
 }
 
-// Watermarking
-function processWatermark(img, watermarkInstructions) {
-    console.log('Processing watermark...');
-    const placement = watermarkInstructions.placement || 'Center';
-    const visibilityPercentage = watermarkInstructions.visibilityPercentage || 100;
-
-    let watermark = watermarkInstructions.watermarkContent;
-    if(watermarkInstructions.isEmbedded){
-        // watermark embedded as base 64
-        console.log('Using embedded watermark');
-        watermark = 'inline:'+watermarkInstructions.watermarkContent;
-    }
-
-    img.dissolve(visibilityPercentage).gravity(placement);
-    img.composite(watermark);
+async function checkJpegSize(outfile, instructions) {
+    // If we specified jpegSize for JPEG output we need to verify that the size does not exceed that
+    if (instructions.fmt === "jpg" && instructions.jpegSize) {
```

Comment:
Use both jpg and jpeg. I some them both being used in worker tests - and fixing the test will time (you have to go look at them all).

Reply:
Good catch. This was in the existing code but seems we might have to adjust.

Aren't we normalizing in api-process or core? If not, shouldn't we do that there?

Related question: currently the output format is picked up by imagemagick purely because of the extension of the output file name, which we set to `fmt`. But there is a `gm().setFormat()` function. I feel we should make it more explicit. wdyt?

Reply:
We are normalizing, but worker tests do not go through api-process (that is where the normalization happens), and they often still use old param names or mix up jpg and jpeg or tif and tiff. 

`But there is a gm().setFormat() function. I feel we should make it more explicit.` <= I think that too. Wasn't sure if you would agree to, so didn't ask before.

Reply:
Actually we do not normalize anything wrt to `jpg` or `jpeg` in api-process. Can't find anything for

- [jpeg in api-process](https://git.corp.adobe.com/nui/api-process/search?utf8=✓&q=jpeg&type=)
- [jpeg in sdk](https://github.com/adobe/asset-compute-sdk/search?q=jpeg&unscoped_q=jpeg)

In the worker selection in `core` we always check both `jpg` and `jpeg`: https://git.corp.adobe.com/nui/core/blob/master/lib/workers.js#L236

Which seems to indicate we rely on workers to handle that themselves and in many cases rely on gm/im support both via the output extension... I think we could improve that and standardize on say `jpg` and normalize for that in api-process. Created [NUI-817](https://jira.corp.adobe.com/browse/NUI-817).

For now I will handle both in the post processing.